luigi/contrib/ftp.py

```diff
@@ -131,14 +131,12 @@ def _sftp_exists(self, path, mtime):
         return exists
 
     def _ftp_exists(self, path, mtime):
-        path_parts = path.split('/')
-        path = '/'.join(path_parts[:-1])
-        fn = path_parts[-1]
+        dirname, fn = os.path.split(path)
 
-        files = self.conn.nlst(path)
+        files = self.conn.nlst(dirname)
 
         exists = False
-        if fn in files:
+        if path in files or fn in files:
```

Comment:
I wonder how universally MDTM is implemented on ftp servers. Our current usage is controlled by a conditional, so users can decide whether to use it.